### PR TITLE
Replace token pool "validate tx" logic stubs with a new "activate" action

### LINKED ISSUE #315: Support for all token pool creation scenarios
# Background

Token pools are initialized with two steps:
1. The actual pool is initialized on a blockchain smart contract
2. The pool details (namespace, name, etc) are announced as a FireFly definition broadcast

The pool creator is responsible for sending 2 only after 1 is confirmed. However, because these steps may or may not occur on the same blockchain, there are multiple possible ways for them to play out.

# Scenarios

## A) 1 and 2 both arrive and are guaranteed to be in that order for all nodes

This is true for the case where tokens and messages are backed by the same blockchain (such as when using Ethereum with ERC1155). Each node may receive and record the blockchain transaction, then fill in the rest of the pool details after the announcement broadcast.

## B) 1 and 2 both arrive, but 2 may arrive before 1

This is possible in cases where tokens and messages are backed by different blockchains. It should be uncommon, but networking conditions (for instance) could cause a particular node to receive 2 quickly while 1 is delayed. There must be a way to save the details from 2 but "rewind" and complete the message processing once 1 is received.

## C) Only 2 arrives, and 1 must be looked up on demand

This is envisioned for cases where the "token pool" is a pre-existing set of tokens (could be on mainnet or a pre-existing private chain). In this case, step 1 for the author involves simply passing details about the existing tokens, but there will be _no new blockchain transaction_ to define the pool. Therefore, the other nodes receiving the broadcast in step 2 must also have a way to lookup and verify the pre-existing token pool.

### LINKED ISSUE #318: Fix inconsistencies between token pools and token transfers
Two inconsistencies in behavior were identified as a side-effect of #315, which should be corrected.

1. The `TokenTransfer` object should store a message ID in addition to the message hash
2. Token pool definition messages should be held in `pending` state until the pool itself is confirmed

### PR DESCRIPTION
**Pull request by awrichar**
Part of #218
Resolves #315
Resolves #318 

A pool must be activated before using it. The activation step will trigger the pool
creation event to be re-sent (so that it can be used to confirm the pool), and will
also trigger the plugin to begin sending transfers from that specific pool.

This solves a number of holes with token pool creation - including validating the
pool announcement against the local plugin and ensuring transfers are not received
before the pool is fully created.

Requires https://github.com/hyperledger/firefly-tokens-erc1155/pull/43 (E2E tests will fail until this is integrated).
There are breaking changes in the internal API between core and the tokens connector,
but they should be invisible to an end user. Migrations are in place to transition any
pre-existing token pools.

### CODE REVIEW

**Review by peterbroadhurst — APPROVED** [review]
Spent a bunch of time staring at `TokenPoolCreated()` and didn't spot any concerns.
The only comment I had was on a method name.

I note the e2e tests are still failing after I updated the manifest and merged with main, so might be one more thing to look at before this clicks in.

**Comment by peterbroadhurst** [comment]
I don't think the e2e failure is related - raising separate issue #328

**Comment by peterbroadhurst** [comment]
https://github.com/kaleido-io/firefly/pull/39 raised as a PR into this branch, to address the (unrelated) E2E failure

**Comment by peterbroadhurst** [comment]
(... the coverage red dot will be addressed when https://github.com/hyperledger/firefly/pull/325 merges)

File: internal/events/token_pool_created.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	"context"
+
+	"github.com/hyperledger/firefly/internal/log"
+	"github.com/hyperledger/firefly/internal/txcommon"
+	"github.com/hyperledger/firefly/pkg/database"
+	"github.com/hyperledger/firefly/pkg/fftypes"
+	"github.com/hyperledger/firefly/pkg/tokens"
+)
+
+func updatePool(storedPool *fftypes.TokenPool, chainPool *tokens.TokenPool) *fftypes.TokenPool {
```
[the remark this answers is not in the document]
Comment (awrichar) [reply internal/events/token_pool_created.go]: Pushed a commit to use `addPoolDetailsFromPlugin`... it's a bit more verbose, but at least it's accurate.

Also removed the return value (switched it to only modify the input param) - so hopefully that makes it clearer as well.

**Pull request merged**